packages/gatsby/cache-dir/app.js

```diff
@@ -184,7 +187,7 @@ apiRunnerAsync(`onClientEntry`).then(() => {
       if (indicatorMountElement) {
         // If user defined replaceHydrateFunction themselves the cleanupFn return might not be there
         // So fallback to unmountComponentAtNode for now
-        if (cleanupFn) {
+        if (cleanupFn && typeof cleanupFn === `function`) {
```

Comment:
Some additional checks here for cases when user/plugin implements `replaceHydrateFunction` and doesn't actually return proper things (unmount function)